test/e2e/storage/generic_persistent_volume-disruptive.go

```diff
@@ -0,0 +1,103 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/api/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+)
+
+var _ = utils.SIGDescribe("GenericPersistentVolume[Disruptive]", func() {
+	f := framework.NewDefaultFramework("generic-disruptive-pv")
+	var (
+		c  clientset.Interface
+		ns string
+	)
+
+	BeforeEach(func() {
+		// Skip tests unless number of nodes is 2
+		framework.SkipUnlessNodeCountIsAtLeast(2)
+		framework.SkipIfProviderIs("local")
+		c = f.ClientSet
+		ns = f.Namespace.Name
+	})
+	disruptiveTestTable := []disruptiveTest{
+		{
+			testItStmt: "Should test that a file written to the mount before kubelet restart is readable after restart.",
+			runTest:    utils.TestKubeletRestartsAndRestoresMount,
+		},
+		{
+			testItStmt: "Should test that a volume mounted to a pod that is deleted while the kubelet is down unmounts when the kubelet returns.",
+			runTest:    utils.TestVolumeUnmountsFromDeletedPod,
+		},
+		{
+			testItStmt: "Should test that a volume mounted to a pod that is force deleted while the kubelet is down unmounts when the kubelet returns.",
+			runTest:    utils.TestVolumeUnmountsFromForceDeletedPod,
+		},
+	}
+	Context("When kubelet restarts", func() {
+		// Test table housing the It() title string and test spec.  runTest is type testBody, defined at
+		// the start of this file.  To add tests, define a function mirroring the testBody signature and assign
+		// to runTest.
+		var (
+			clientPod *v1.Pod
+			pvc       *v1.PersistentVolumeClaim
+		)
+		BeforeEach(func() {
+			framework.Logf("Initializing pod and pvcs for test")
+			clientPod, pvc = createPodPVCFromSC(f, c, ns)
+		})
+		for _, test := range disruptiveTestTable {
+			func(t disruptiveTest) {
+				It(t.testItStmt, func() {
+					By("Executing Spec")
+					t.runTest(c, f, clientPod, pvc)
+				})
+			}(test)
+		}
+		AfterEach(func() {
+			framework.Logf("Tearing down test spec")
+			tearDownTestCase(c, f, ns, clientPod, pvc, nil)
+			pvc, clientPod = nil, nil
+		})
+	})
+})
+
+func createPodPVCFromSC(f *framework.Framework, c clientset.Interface, ns string) (*v1.Pod, *v1.PersistentVolumeClaim) {
+	var err error
+	test := storageClassTest{
+		name:      "default",
+		claimSize: "2Gi",
+	}
+	pvc := newClaim(test, ns, "default")
+	pvc, err = c.CoreV1().PersistentVolumeClaims(pvc.Namespace).Create(pvc)
+	Expect(err).NotTo(HaveOccurred(), "Error creating pvc")
+	pvcClaims := []*v1.PersistentVolumeClaim{pvc}
+	pvs, err := framework.WaitForPVClaimBoundPhase(c, pvcClaims, framework.ClaimProvisionTimeout)
```

Comment:
why use a slice here? Is it possible to call `framework.WaitForPersistentVolumeClaimPhase`?

Reply:
does it matter? I notice that `WaitForPVClaimBoundPhase` internally calls `WaitForPersistentVolumeClaimPhase`

Reply:
only in that you would not need to check for the len slice == 1 below.
Otherwise it is just my observation.